ui/pages/confirmations/components/signature-request/signature-request.js

```diff
@@ -208,6 +214,25 @@ const SignatureRequest = ({
     primaryType,
   } = parseMessage(data);
 
+  const hasMigratedFromOpenSeaToBlockaid = useSelector(
+    getHasMigratedFromOpenSeaToBlockaid,
+  );
+  const isNetworkSupportedByBlockaid = useSelector(
+    getIsNetworkSupportedByBlockaid,
+  );
+  const hasDismissedOpenSeaToBlockaidBanner = useSelector(
+    getHasDismissedOpenSeaToBlockaidBanner,
+  );
+
+  const showOpenSeaToBlockaidBannerAlert =
+    hasMigratedFromOpenSeaToBlockaid &&
+    !isNetworkSupportedByBlockaid &&
+    !hasDismissedOpenSeaToBlockaidBanner;
+
+  const handleCloseOpenSeaToBlockaidBannerAlert = () => {
+    dispatch(dismissOpenSeaToBlockaidBanner());
+  };
```

Comment:
Q: have we considered moving this logic into BlockaidBannerAlert itself? 

we could return null if !showOpenSeaToBlockaidBannerAlert

Reply:
That's a great suggestion, but perhaps we can consider this improvement separately from this PR since it's a release blocker. Thanks!